server/src/main/java/io/crate/metadata/doc/DocSchemaInfo.java

```diff
@@ -284,6 +288,81 @@ public void update(ClusterChangedEvent event) {
                 schemaName,
                 udfMetadata.functionsMetadata().stream().filter(f -> schemaName.equals(f.schema())));
         }
+
+        PublicationsMetadata prevPublicationsMetadata = prevMetadata.custom(PublicationsMetadata.TYPE);
+        PublicationsMetadata newPublicationsMetadata = newMetadata.custom(PublicationsMetadata.TYPE);
+        var updatedTables = updatePublicationState(prevPublicationsMetadata, newPublicationsMetadata, docTableByName);
+        for (String updatedTable : updatedTables) {
+            docTableByName.remove(updatedTable);
+        }
+    }
+
+    @VisibleForTesting
+    static Set<String> updatePublicationState(@Nullable PublicationsMetadata prevMetadata,
+                                              @Nullable PublicationsMetadata newMetadata,
+                                              Map<String, DocTableInfo> docTableByName) {
```

Comment:
I think this could use a different name. Maybe something like `getTablesAffectedByPublicationsChange`. The current name seems missleading as it doesn't update the publications state.